main.nf

```diff
@@ -92,44 +53,671 @@ if (workflow.profile.contains('awsbatch')) {
+        git lfs pull -I public --include "data*clinical*sample.txt"
+        git lfs pull -I public --include "data_mutations_mskcc.txt"
+        cd ..
+        cat datahub/public/*/data_mutations_mskcc.txt > cbioportal_allstudies_data_mutations_mskcc.txt
+        cat datahub/public/*/*data*clinical*sample.txt | \\
+            awk 'BEGIN{FS=OFS="\\t"}{if(\$1!~"#SAMPLE_ID"){gsub("#SAMPLE_ID", "\\nSAMPLE_ID");} print}' | \\
+            awk 'BEGIN{FS=OFS="\\t"}{s=0; j=0; \\
+            for(i=1;i<=NF;i++){ \\
+               if(\$i=="CANCER_TYPE_DETAILED") j=1; \\
+               if(\$i=="CANCER_TYPE") s=1;} \\
+            if(j==1 && s==0){ \\
+              gsub("CANCER_TYPE_DETAILED", "CANCER_TYPE");} print;}' \\
+            > cbioportal_allstudies_data_clinical_sample.txt
+        """
+    else
+        """
+        pypgatk_cli.py cbioportal-downloader --config_file "${cbioportal_config}" -d "${params.cbioportal_study_id}"
+        tar -xzvf database_cbioportal/${params.cbioportal_study_id}.tar.gz
+        cat ${params.cbioportal_study_id}/data_mutations_mskcc.txt > cbioportal_allstudies_data_mutations_mskcc.txt
+        cat ${params.cbioportal_study_id}/data_clinical_sample.txt | \\
+            awk 'BEGIN{FS=OFS="\\t"}{if(\$1!~"#SAMPLE_ID"){gsub("#SAMPLE_ID", "\\nSAMPLE_ID");} print}' | \\
+            awk 'BEGIN{FS=OFS="\\t"}{s=0; j=0; \\
+            for(i=1;i<=NF;i++){ \\
+               if(\$i=="CANCER_TYPE_DETAILED") j=1; if(\$i=="CANCER_TYPE") s=1;} \\
+            if(j==1 && s==0){gsub("CANCER_TYPE_DETAILED", "CANCER_TYPE");} print;}' \\
+            > cbioportal_allstudies_data_clinical_sample.txt
+        """
+ }
+
+/**
+ * Generate proteinDB from cBioPortal mutations
+ */
+ process cbioportal_proteindb{
+
+   publishDir "${params.outdir}", mode: 'copy', overwrite: true
+
+   when:
+   params.cbioportal
+
+   input:
+   file g from ch_GRCh37_cds
+   file m from cbio_mutations
+   file s from cbio_samples
+   file cbioportal_config
+
+   output:
+   file 'cbioPortal_proteinDB*.fa' into cBioportal_proteindb
+
+   script:
+   """
+   pypgatk_cli.py cbioportal-to-proteindb --config_file ${cbioportal_config} --input_mutation ${m} --input_cds ${g} \\
+                  --clinical_sample_file ${s} --filter_column ${params.cbioportal_filter_column} \\
+                  --accepted_values ${params.cbioportal_accepted_values} --output_db cbioPortal_proteinDB.fa
+   """
+}
+
+merged_databases = merged_databases.mix(cBioportal_proteindb)
+
+/**
+ * Concatenate all generated databases from merged_databases channel to the final_database_protein file
+ */
+process merge_proteindbs {
+
+   publishDir "${params.outdir}", mode: 'copy', overwrite: true
+
+   input:
+   file("proteindb*") from merged_databases.collect()
+
+   output:
+   file 'merged_databases.fa' into to_clean_ch
+
+   script:
+   """
+   cat proteindb* > merged_databases.fa
+   """
+}
+
+stop_codons = ''
+if (params.add_stop_codons){
+	stop_codons = "--add_stop_codons"
+}
```

Comment:
I think it would be better to have this in the script block of `clean_protein_database`